pkg/lookups/lookups.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package lookups
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/siglens/siglens/pkg/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/valyala/fasthttp"
+)
+
+const (
+	lookupsDir      = "lookups"
+	allowedExtCSV   = ".csv"
+	allowedExtCSVGZ = ".csv.gz"
+)
+
+func UploadLookupFile(ctx *fasthttp.RequestCtx) {
+	fileName := string(ctx.FormValue("name"))
+	if fileName == "" {
+		log.Error("UploadLookupFile: File name is required")
+		ctx.Error("File name is required", fasthttp.StatusBadRequest)
+		return
+	}
+
+	fileHeader, err := ctx.FormFile("file")
+	if err != nil {
+		log.Errorf("UploadLookupFile: Error retrieving the file: %v", err)
+		ctx.Error("Error retrieving the file", fasthttp.StatusBadRequest)
+		return
+	}
+
+	// Check for .csv and .csv.gz extensions
+	lowerFileName := strings.ToLower(fileHeader.Filename)
+	isCSV := strings.HasSuffix(lowerFileName, allowedExtCSV)
+	isCSVGZ := strings.HasSuffix(lowerFileName, allowedExtCSVGZ)
+
+	if !isCSV && !isCSVGZ {
+		log.Errorf("UploadLookupFile: Invalid file type: %s", filepath.Ext(fileHeader.Filename))
+		ctx.Error(fmt.Sprintf("Invalid file type. Only %s and %s files are allowed", allowedExtCSV, allowedExtCSVGZ), fasthttp.StatusBadRequest)
+		return
+	}
+
+	if !strings.HasSuffix(strings.ToLower(fileName), allowedExtCSV) &&
+		!strings.HasSuffix(strings.ToLower(fileName), allowedExtCSVGZ) {
+		if isCSVGZ {
+			fileName += allowedExtCSVGZ
+		} else {
+			fileName += allowedExtCSV
+		}
+	}
+
+	fullLookupsDir := filepath.Join(config.GetDataPath(), lookupsDir)
```

Comment:
There's already a function for this:
```suggestion
	fullLookupsDir := config.GetLookupPath()
```

And then I think you can delete the `lookupsDir` global constant